src/javascript/app/pages/trade/markets/market-selector.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import {
+    BreakpointProvider,
+    CustomDropdown,
+    useDropdown,
+} from '@deriv-com/quill-ui';
+import { MarketsDropdown, getMarketName } from './markets-dropdown.jsx';
+import { getElementById } from '../../../../_common/common_functions';
+import { localize } from '../../../../_common/localize';
+import { useContractChange, useMarketChange } from '../../../hooks/events.js';
+import Defaults, { PARAM_NAMES } from '../defaults.js';
+
+const Contracts = () => {
+    const { close } = useDropdown();
+    useEffect(() => {
+        document
+            .querySelectorAll('.contract-item-clickables')
+            .forEach((element) => {
+                element.addEventListener('click', () => {
+                    close();
+                });
+            });
+    }, []);
+
+    return <div id='contract-injection-container' />;
+};
+
+const MarketSelector = () => {
+    const [marketLabel, setMarketLabel] = useState(getMarketName());
+    const [tradeTypeLabel, setTradeTypeLabel] = useState('');
+    const hasContractChange = useContractChange();
+    const hasMarketChange = useMarketChange();
+
+    const observeContractDropdown = () => {
+        const contractItem = document.getElementById('contract_component');
+        const observer = new MutationObserver((mutations) => {
+            mutations.forEach((mutation) => {
+                const parentElement = document.querySelector(
+                    '.quill-custom-contract-dropdown .quill-custom-dropdown__content'
+                );
+
+                mutation.removedNodes.forEach((node) => {
+                    if (node === parentElement) {
+                        document
+                            .getElementById('tradetype-dropdown')
+                            .appendChild(contractItem);
+                    }
+                });
+
+                mutation.addedNodes.forEach((node) => {
+                    if (node === parentElement) {
+                        const customDropdownContainer = document.getElementById(
+                            'contract-injection-container'
+                        );
+
+                        if (customDropdownContainer) {
+                            customDropdownContainer.appendChild(contractItem);
+                        }
+                    }
+                });
+            });
+        });
```

Comment:
This change is essential for handling the contracts dropdown. It automatically injects the current contract dropdown into the custom dropdown and action sheet when they are opened.

@meenakshi-deriv @jia-deriv @Nuzhy-Deriv Please note: If the class quill-custom-dropdown__content is removed from the Input Dropdown during future refactoring, this code will need to be updated accordingly.